### PR TITLE
Do not select any item by default

### PR DESCRIPTION
**Pull request by manolo**
Connected to https://github.com/vaadin/vaadin-core/issues/105

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-list-mixin/6)

### CODE REVIEW

**Comment by tomivirkki** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---

*[vaadin-list-mixin.html, line 24 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-list-mixin/6#-Kwm70hHFYX63UQUJn1W:-Kwm70hHFYX63UQUJn1X:bfv4rta) ([raw file](https://github.com/vaadin/vaadin-list-mixin/blob/a199f6ff0d818163bfd7d5b2b672e1c291d81e70/vaadin-list-mixin.html#L24)):*
> ```HTML
>         selected: {
>           type: Number,
>           value: -1,
> ```

Why `-1` and not just leaving this undefined?

---

*[vaadin-list-mixin.html, line 132 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-list-mixin/6#-Kwm7AGk_veefZ-fBIZ0:-Kwm7AGk_veefZ-fBIZ1:bxtgtce) ([raw file](https://github.com/vaadin/vaadin-list-mixin/blob/a199f6ff0d818163bfd7d5b2b672e1c291d81e70/vaadin-list-mixin.html#L132)):*
> ```HTML
>     _setFocusable(idx) {
>       const items = this._navItems;
>       const item = items[idx] || items[0];
> ```

Could use a test

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-list-mixin/6)*

**Comment by manolo** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---

*[vaadin-list-mixin.html, line 24 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-list-mixin/6#-Kwm70hHFYX63UQUJn1W:-KwnJZ03uQ7F9u8Cl0uI:bvl7xc0) ([raw file](https://github.com/vaadin/vaadin-list-mixin/blob/a199f6ff0d818163bfd7d5b2b672e1c291d81e70/vaadin-list-mixin.html#L24)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Why `-1` and not just leaving this undefined?
</blockquote></details>

Just because is declared as a number, but yes undefined or any thing that does not return any item from the array also works.
Do I remove the value line or set `value: undefined ` to be more explicit 

---

*[vaadin-list-mixin.html, line 132 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-list-mixin/6#-Kwm7AGk_veefZ-fBIZ0:-KwnK9jxueaCVFcnuw9-:b-q7gusm) ([raw file](https://github.com/vaadin/vaadin-list-mixin/blob/a199f6ff0d818163bfd7d5b2b672e1c291d81e70/vaadin-list-mixin.html#L132)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Could use a test
</blockquote></details>

You mean test this case? Actually we already have tests for both cases `should set tabIndex=-1 to all items, but the first`  and  `should move tabIndex when moving focus`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-list-mixin/6)*

**Comment by tomivirkki** [comment]
*[vaadin-list-mixin.html, line 24 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-list-mixin/6#-Kwm70hHFYX63UQUJn1W:-KwnUoocw_h1840B9zpj:bqqgcfy) ([raw file](https://github.com/vaadin/vaadin-list-mixin/blob/a199f6ff0d818163bfd7d5b2b672e1c291d81e70/vaadin-list-mixin.html#L24)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Just because is declared as a number, but yes undefined or any thing that does not return any item from the array also works.
Do I remove the value line or set `value: undefined ` to be more explicit 
</blockquote></details>

Removing the value should be enough. lgtm

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-list-mixin/6)*

**Comment by tomivirkki** [comment]
*[vaadin-list-mixin.html, line 132 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-list-mixin/6#-Kwm7AGk_veefZ-fBIZ0:-KwnVM9oAuEoFO1l2RUv:b-5ree7f) ([raw file](https://github.com/vaadin/vaadin-list-mixin/blob/a199f6ff0d818163bfd7d5b2b672e1c291d81e70/vaadin-list-mixin.html#L132)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

You mean test this case? Actually we already have tests for both cases `should set tabIndex=-1 to all items, but the first`  and  `should move tabIndex when moving focus`
</blockquote></details>

I meant a test case for the new added ` || items[0]` part. Something like `it('should set the first item focusable by default')`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-list-mixin/6)*

**Comment by manolo** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-list-mixin.html, line 132 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-list-mixin/6#-Kwm7AGk_veefZ-fBIZ0:-KwnYpjaOh86fujxFW19:b-prvy5w) ([raw file](https://github.com/vaadin/vaadin-list-mixin/blob/a199f6ff0d818163bfd7d5b2b672e1c291d81e70/vaadin-list-mixin.html#L132)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

I meant a test case for the new added ` || items[0]` part. Something like `it('should set the first item focusable by default')`
</blockquote></details>

Fixed the test that was testing this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-list-mixin/6)*

File: vaadin-list-mixin.html [hunk 1]
```diff
@@ -129,7 +128,8 @@
 
     _setFocusable(idx) {
```
Comment (abdonrd) [inline vaadin-list-mixin.html]: `idx`? Maybe `index` is better?

Same in `_focus(inx)`.

**Comment by tomivirkki** [comment]
:lgtm: 

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-list-mixin/6#-:-Kwo2wtaBFBdhVa_n7-3:bg7igkv)*

**Comment by YuriyVaadin** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-list-mixin/6#-:-Kwo4lDre5Ix6hmEEgwk:bnfp4nl)*

**Pull request merged**